test/e2e/traces_test.go

```diff
@@ -0,0 +1,244 @@
+		"{{OBS_CLIENT_SECRET}}",
+		clientSecret, -1)
+	forwardingConfig = strings.Replace(forwardingConfig,
+		"{{OIDC_TOKEN_URL}}",
+		tokenUrl, -1)
+	forwardingConfig = strings.Replace(forwardingConfig,
+		"{{OBS_GRPC_ENDPOINT}}",
+		api.InternalEndpoint("grpc"), -1)
+	forwardingConfig = strings.Replace(forwardingConfig,
+		"{{DEX_TOKEN}}",
+		token, -1)
+
+	otelFile, err := ioutil.TempFile(e.SharedDir(), "fwd-coll.yaml")
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+
+	defer os.Remove(otelFile.Name())
+
+	_, err = otelFile.Write([]byte(forwardingConfig))
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+
+	// otelFile.Name() will give relative pathname to temp file.  Docker will complain with
+	// "If you intended to pass a host directory, use absolute path.""
+	otelFileName, err := filepath.Abs(otelFile.Name())
+	if err != nil {
+		t.Fatalf("unexpected Abs() error: %s", err)
+	}
+
+	otel := e.Runnable("otel-fwd-coll").
+		WithPorts(
+			map[string]int{
+				"http": 4318,
+				"grpc": 4317,
+			}).
+		Init(e2e.StartOptions{
+			// Note that if the forwarding collector was OIDC flow we would need
+			// "otel/opentelemetry-collector-contrib:0.45.0" instead.
+			Image:   "otel/opentelemetry-collector:0.45.0",
+			Volumes: []string{fmt.Sprintf("%s:/conf/collector.yaml", otelFileName)},
+			Command: e2e.Command{
+				Args: []string{"--config=/conf/collector.yaml"},
+			},
+		})
+
+	testutil.Ok(t, e2e.StartAndWaitReady(otel))
+
+	// Send trace insecurly to forwarding OTel collector for forwarding through Observatorium
+	// (This code could be refactored to observatorium/up, the test client)
+
+	client := &http.Client{}
+	request, err := http.NewRequest(
+		"POST",
+		fmt.Sprintf("http://%s/v1/traces", otel.Endpoint("http")),
+		bytes.NewBuffer([]byte(traceJSON)))
+	testutil.Ok(t, err)
+	request.Header.Set("Content-Type", "application/json")
+	response, err := client.Do(request)
+	testutil.Ok(t, err)
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		responseBytes, _ := io.ReadAll(response.Body)
+		t.Fatalf("forwarding trace to OTel collector not OK: %d\n%s", response.StatusCode, responseBytes)
+	}
+
+	request, err = http.NewRequest(
+		"GET",
+		fmt.Sprintf("http://%s/api/v3/traces/%s", httpQueryEndpoint, "5B8EFFF798038103D269B633813FC60C"),
+		nil)
+	testutil.Ok(t, err)
+
+	// Note that we don't wait for the trace to be committed to storage.
+	// (If we were using a buffered Jaeger storage backend we would
+	// not give up on the first fetch attempt.)
+	response, err = client.Do(request)
+	testutil.Ok(t, err)
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		responseBytes, _ := io.ReadAll(response.Body)
+		t.Fatalf("reading trace from Jaeger not OK: %d\n%s", response.StatusCode, responseBytes)
+	}
+	// Parse and verify the trace is the same?
+	// responseBytes, _ := io.ReadAll(response.Body)
+	// ...
+
+	// Uncomment for interactive test
```

Comment:
Might be a good idea to add tracing setup to `TestInteractiveSetup`, as was done for Rules API in https://github.com/observatorium/api/pull/228. This can be a separate PR too I think. 🙂